### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by JulienDoerner**
This PR contians the update on the documentation of the source features. Also the example notebooks for the custom photon field and the diffusion in the Milky Way are added.

### CODE REVIEW

File: include/crpropa/Source.h [hunk 1]
```diff
@@ -606,7 +662,10 @@ class SourceDirection: public SourceFeature {
 
 /**
  @class SourceEmissionMap
- @brief Deactivate Candidate if it has zero probability in provided EmissionMap
+ @brief Deactivate Candidate if it has zero probability in provided EmissionMap. 
+
+	This feature does not change the direction of the candidate. Therefore a usefull direction feature (isotropic or directed emission)
+	must be added to the sources before. The propability of the emission map is not taken into account.
```
Comment (lukasmerten) [inline include/crpropa/Source.h:668]: What does "the probability of the emission map is not taken into account" mean?

Comment (JulienDoerner) [reply include/crpropa/Source.h:668]: it is only a filtering for p=0 or not.

Comment (lukasmerten) [reply include/crpropa/Source.h:668]: You mean that the emission map does not make a difference between emission probability p=1 or p=0.00001? 
The only parameter taken into account is (p>1) == True?

In that case I would reformulate to: "A weighting of the source emission directions is not included. Candidates are either emitted with a weight=1 for or not injected into the simulation."

Comment (lukasmerten) [reply include/crpropa/Source.h:668]: Anyway, I guess it's fine. I might add some additional comment later.

**Review by lukasmerten — APPROVED** [review]

**Pull request merged**